dbms/include/DB/Functions/FunctionsStringSearch.h

```diff
@@ -1061,7 +1061,7 @@ struct ReplaceStringImpl
 			bool can_finish_current_string = false;
 
 			/// Проверяем, что вхождение не переходит через границы строк.
-			if (match + needle.size() < begin + n * (i + 1))
+			if (match + needle.size() - 1 < begin + n * (i + 1))
```

Comment:
Put `clickhouse` program to `/usr/bin` (or PATH);
make symlink `clickhouse-client` in `/usr/bin` pointing to `clickhouse`*.

Run clickhouse-server.

Change directory to `dbms/tests/` in source tree
and run `./clickhouse-test`.

*- Also you could just build (`./release --standalone`) and install debian packages.